evap/middleware.py

```diff
@@ -0,0 +1,30 @@
+from functools import wraps
+
+from django.conf import settings
+from django.shortcuts import redirect
+
+class RequireLoginMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        return self.get_response(request)
+
+    @staticmethod
+    def process_view(request, view_func, _view_args, _view_kwargs):
+        # Returning None tells django to pass the request on
+        if request.user.is_authenticated:
+            return None
+
+        if "no_login_required" in view_func.__dict__ and view_func.no_login_required:
+            return None
+
+        return redirect(f"{settings.LOGIN_URL}?next={request.path}")
+
+
+def no_login_required(func):
+    func.no_login_required = True
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
```

Comment:
I wonder if these three lines are even required here - we don't want to add any additional code in the wrapper before or after the call, nor do we want to change arguments. Would simply `return func` also do it?

Reply:
Wrapper is not needed. The decorator `evaluation.auth.user_passes_text` used to check for an authenticated user, this is gone now. I couldn't find any other redirects to the login page.

As for tests: Three auth tests failed initially, but I fixed that by adding `no_login_required` to `login_key_authentication`.